### ISSUE TITLE
paths in jsconfig.json not working in insider

### ISSUE THREAD

**Issue by zaaack** [Microsoft/vscode#34928]
- VSCode Version: 1.17.0-insider f7962f0682a76516df51de4856f8ccc5d8ad578a
- OS Version: macOS 10.12.3

Steps to Reproduce:

1. open a js project with jsconfig.json:

```json
{
    "compilerOptions": {
        "jsx": "react",
        "module": "commonjs",
        "allowSyntheticDefaultImports": true,
        // This must be specified if "paths" is set
        "baseUrl": "./",
        "moduleResolution": "node",
        // Relative to "baseUrl"
        "paths": {
            "common/*": ["src/common/*"]
        }
    }
}

```

then press F3 to go to definition, then failed with cannot find any definition for xxx.

Same project will work in vscode-stable.

2.

Reproduces without extensions: No

It's strange that after it worked if I run with `--disable-extensions`, but I cannot find which extension cause this.

```sh
z at ZsMBP in ~/Projects/Rust/aria2c-gui (master) 
$ code --list-extensions                      
DavidAnson.vscode-markdownlint
EditorConfig.EditorConfig
HookyQR.beautify
Ionide.Ionide-FAKE
Ionide.Ionide-fsharp
Ionide.Ionide-Paket
PeterJausovec.vscode-docker
Shan.code-settings-sync
Zignd.html-css-class-completion
agauniyal.vscode-caniuse
alphabotsec.vscode-eclipse-keybindings
anseki.vscode-color
christian-kohler.npm-intellisense
christian-kohler.path-intellisense
dbaeumer.vscode-eslint
donjayamanne.githistory
donjayamanne.python
eg2.tslint
file-icons.file-icons
formulahendry.code-runner
freebroccolo.reasonml
hackwaly.ocaml
humao.rest-client
jrieken.backspace-plusplus
jtlowe.vscode-icon-theme
kogai.regex-railroad-diagrams
kumar-harsh.graphql-for-vscode
ms-vscode.csharp
qinjia.seti-icons
robertohuertasm.vscode-icons
robinbentley.sass-indented
rust-lang.rust
searKing.preview-vscode
steoates.autoimport
vadimcn.vscode-lldb
wix.vscode-import-cost
wmaurer.vscode-jumpy
xabikos.JavaScriptSnippets
xyz.local-history

```

**Comment by mjbvz** [comment]
Sounds like one of those extensions is to blame. Try disabling all extensions that apply to JavaScript or TypeScript and then enabling them one by one to track down which one may have caused this

You may also want to try [changing the active TypeScript version](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions) in your workspace to see if this fixes things

**Comment by zaaack** [comment]
@mjbvz I tried replacing typescript version to global installed 2.3.4, it worked! Looks like typescript in vscode-insider cannot work well with current version of jsconfig.json.

**Comment by mjbvz** [comment]
@zaaack Can you please also try this with TypeScript 2.5.3 which is included in VSCode 1.17?

If that doesn't work, also try out the nightly typescript as well (`npm install typescript@next`) to see if the issue has been fixed there in that
